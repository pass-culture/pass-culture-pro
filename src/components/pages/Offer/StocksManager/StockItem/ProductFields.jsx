import moment from 'moment'
import { assignModalConfig } from 'pass-culture-shared'
import PropTypes from 'prop-types'
import React, { Component, Fragment } from 'react'
import { createParseNumberValue } from 'react-final-form-utils'
import ReactTooltip from 'react-tooltip'

import {
  createFormatAvailable,
  createFormatPrice,
  getRemainingStocksCount,
} from './utils'
import { DateField, HiddenField, NumberField } from 'components/layout/form'
import Icon from 'components/layout/Icon'

export class ProductFields extends Component {
  static isParsedByForm = true

  constructor() {
    super()
    this.isPriceInputDeactivate = false
  }

  componentDidUpdate() {
    ReactTooltip.rebuild()
  }

  componentWillUnmount() {
    this.props.closeInfo()
  }

  onPriceBlur = event => {
    if (this.isPriceInputDeactivate) {
      return
    }

    const { closeInfo, dispatch, hasIban, readOnly, showInfo } = this.props
    const formPrice = createParseNumberValue('number')(event.target.value)

    if (readOnly || hasIban || !formPrice) {
      return
    }

    const priceInput = event.target
    priceInput.focus()
    this.isPriceInputDeactivate = true

    dispatch(assignModalConfig({ extraClassName: 'modal-in-modal' }))

    showInfo(
      <Fragment>
        <div className="mb24 has-text-left">
          Vous avez saisi une offre payante. Pensez à demander à
          l'administrateur financier nommé pour votre structure de renseigner
          son IBAN. Sans IBAN, les réservations de vos offres éligibles ne vous
          seront pas remboursées.
        </div>
        <div className="has-text-right">
          <button
            className="button is-primary"
            onClick={() => {
              dispatch(assignModalConfig({ extraClassName: null }))
              closeInfo()
              this.isPriceInputDeactivate = false
            }}>
            J'ai compris
          </button>
        </div>
      </Fragment>
    )
  }

  render() {
    const { beginningDatetime, isEventStock, readOnly, stock } = this.props
    const { available, bookings } = stock || {}
    const remainingStocksCount = getRemainingStocksCount(
      available,
      bookings || []
    )

    return (
      <Fragment>
        <td title="Gratuit si vide">
          <HiddenField name="offerId" type="hidden" />
          <HiddenField name="venueId" type="hidden" />
          <NumberField
            name="price"
            onBlur={this.onPriceBlur}
            format={createFormatPrice(readOnly)}
            placeholder="Gratuit"
            readOnly={readOnly}
            title="Prix"
          />
        </td>
        <td title="Laissez vide si pas de limite">
          <DateField
            maxDate={isEventStock ? moment(beginningDatetime) : undefined}
            name="bookingLimitDatetime"
            placeholder="Laissez vide si pas de limite"
            readOnly={readOnly}
          />
        </td>
        <td className="tooltiped">
          <NumberField
            format={createFormatAvailable(readOnly)}
            name="available"
            placeholder="Illimité"
            readOnly={readOnly}
            renderValue={() => {
              if (readOnly) {
                return null
              }
              return (
                <span
                  className="button tooltip qty-info"
                  data-place="bottom"
                  data-tip="<p>Laissez ce champ vide pour un nombre de places ou stock illimité.</p>"
                  data-type="info">
                  <Icon svg="picto-info" />
                </span>
              )
            }}
            title="Stock[ou] Places affecté[es]"
          />
        </td>

        <td className="is-small remaining-stock" id="remaining-stock">
          {remainingStocksCount}
        </td>
      </Fragment>
    )
  }
}

ProductFields.defaultProps = {
  beginningDatetime: null,
  readOnly: true,
  isEventStock: true,
  offer: null,
  stock: null,
}

ProductFields.propTypes = {
  beginningDatetime: PropTypes.string,
  closeInfo: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
  isEventStock: PropTypes.bool,
  offer: PropTypes.object,
  parseFormChild: PropTypes.func,
  readOnly: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  showInfo: PropTypes.func.isRequired,
  stock: PropTypes.object,
}

export default ProductFields
