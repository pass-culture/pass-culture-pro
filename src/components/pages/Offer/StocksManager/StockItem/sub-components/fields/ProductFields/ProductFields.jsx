import moment from 'moment'
import PropTypes from 'prop-types'
import React, { PureComponent, Fragment } from 'react'
import { createParseNumberValue } from 'react-final-form-utils'
import ReactTooltip from 'react-tooltip'

import { createFormatAvailable, formatPrice, getRemainingStocksCount } from '../../../utils/utils'
import DateField from '../../../../../../../layout/form/fields/DateField'
import HiddenField from '../../../../../../../layout/form/fields/HiddenField'
import NumberField from '../../../../../../../layout/form/fields/NumberField'
import Icon from '../../../../../../../layout/Icon'

class ProductFields extends PureComponent {
  componentDidUpdate() {
    ReactTooltip.rebuild()
  }

  componentWillUnmount() {
    const { closeInfo } = this.props
    closeInfo()
  }

  handleOnPriceBlur = event => {
    const { assignModalConfig, hasIban, readOnly, showInfo } = this.props
    const formPrice = createParseNumberValue('number')(event.target.value)

    if (readOnly || hasIban || !formPrice) {
      return
    }

    event.target.focus()
    assignModalConfig('modal-in-modal')

    showInfo(
      <Fragment>
        <div className="mb24 has-text-left">
          {'Vous avez saisi une offre payante. Pensez à demander à '}
          {'l’administrateur financier nommé pour votre structure de renseigner '}
          {'son IBAN. Sans IBAN, les réservations de vos offres éligibles ne vous '}
          {'seront pas remboursées.'}
        </div>
        <div className="has-text-right">
          <button
            className="button is-primary"
            onClick={this.handleOnClick}
            type="button"
          >
            {'J’ai compris'}
          </button>
        </div>
      </Fragment>
    )
  }

  handleOnClick = () => {
    const { assignModalConfig, closeInfo } = this.props
    assignModalConfig(null)
    closeInfo()
  }

  renderDateFieldValue = (readOnly, venue) => () => {
    if (readOnly) {
      return null
    }

    let tip = "L'heure limite de réservation de ce jour est 23h59 et ne peut pas être changée."
    if (venue && venue.isVirtual) {
      tip = `${tip}<br /><br />Vous êtes sur une offre numérique, la zone horaire de cette date correspond à celle de votre structure.`
    }

    return (
      <span
        className="button tooltip tooltip-info"
        data-place="bottom"
        data-tip={`<p>${tip}</p>`}
      >
        <Icon svg="picto-info" />
      </span>
    )
  }

  renderNumberFieldValue = readOnly => () => {
    if (readOnly) {
      return null
    }
    return (
      <span
        className="button tooltip tooltip-info"
        data-place="bottom"
        data-tip="<p>Laissez ce champ vide pour un nombre de places ou stock illimité.</p>"
      >
        <Icon svg="picto-info" />
      </span>
    )
  }

  render() {
    const { beginningDatetime, isEvent, readOnly, stock, timezone, venue } = this.props
    const { available, remainingQuantity } = stock || {}
    const remainingStocksCount = getRemainingStocksCount(available, remainingQuantity)

    return (
      <Fragment>
        <td title="Gratuit si vide">
          <HiddenField
            name="offerId"
            type="hidden"
          />
          <HiddenField
            name="venueId"
            type="hidden"
          />
          <NumberField
            format={formatPrice(readOnly)}
            min="0"
            name="price"
            onBlur={this.handleOnPriceBlur}
            placeholder="Gratuit"
            readOnly={readOnly}
            title="Prix"
          />
        </td>
        <td
          className="tooltiped"
          title="Laissez vide si pas de limite"
        >
          <DateField
            maxDate={isEvent ? moment(beginningDatetime) : undefined}
            name="bookingLimitDatetime"
            placeholder="Laissez vide si pas de limite"
            readOnly={readOnly}
            renderValue={this.renderDateFieldValue(readOnly, venue)}
            timezone={timezone}
          />
        </td>
        <td className="tooltiped">
          <NumberField
            format={createFormatAvailable()}
            min="0"
            name="available"
            placeholder="Illimité"
            readOnly={readOnly}
            renderValue={this.renderNumberFieldValue(readOnly)}
            title="Stock[ou] Place[s] affecté[es]"
          />
        </td>

        <td
          className="is-small remaining-stock"
          id="remaining-stock"
        >
          {remainingStocksCount}
        </td>
      </Fragment>
    )
  }
}

ProductFields.defaultProps = {
  beginningDatetime: null,
  isEvent: true,
  readOnly: true,
  stock: null,
  timezone: null,
}

ProductFields.propTypes = {
  assignModalConfig: PropTypes.func.isRequired,
  beginningDatetime: PropTypes.string,
  closeInfo: PropTypes.func.isRequired,
  hasIban: PropTypes.bool.isRequired,
  isEvent: PropTypes.bool,
  readOnly: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  showInfo: PropTypes.func.isRequired,
  stock: PropTypes.shape(),
  timezone: PropTypes.string,
  venue: PropTypes.shape().isRequired,
}

export default ProductFields
