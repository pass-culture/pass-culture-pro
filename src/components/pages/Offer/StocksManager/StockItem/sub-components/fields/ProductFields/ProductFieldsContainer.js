import { assignModalConfig } from 'pass-culture-shared'
import { connect } from 'react-redux'

import ProductFields from './ProductFields'

export const mapDispatchToProps = dispatch => {
  return {
    assignModalConfig: (extraClassName) => {
      dispatch(
        assignModalConfig(
          {extraClassName}
        )
      )
    }
  }
}

export default connect(
  null,
  mapDispatchToProps
)(ProductFields)
