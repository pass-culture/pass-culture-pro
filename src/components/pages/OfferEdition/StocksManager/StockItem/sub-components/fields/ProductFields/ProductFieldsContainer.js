import { connect } from 'react-redux'
import ProductFields from './ProductFields'
import { assignModalConfig } from 'pass-culture-shared'

export const mapDispatchToProps = dispatch => {
  return {
    assignModalConfig: (extraClassName) => {
      dispatch(
        assignModalConfig(
          {extraClassName}
        )
      )
    }
  }
}

export default connect(
  null,
  mapDispatchToProps
)(ProductFields)
