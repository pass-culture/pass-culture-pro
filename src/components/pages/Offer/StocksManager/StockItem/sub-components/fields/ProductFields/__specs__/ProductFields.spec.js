import React from 'react'
import { shallow } from 'enzyme'

import ProductFields from '../ProductFields'

describe('src | components | pages | Offer | StocksManager | StockItem | sub-components | fields | ProductFields', () => {
  it('should match the snapshot', () => {
    // given
    const initialProps = {
      closeInfo: jest.fn(),
      dispatch: jest.fn(),
      hasIban: false,
      parseFormChild: jest.fn(),
      showInfo: jest.fn(),
      venue: {},
    }

    // when
    const wrapper = shallow(<ProductFields {...initialProps} />)

    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
})
