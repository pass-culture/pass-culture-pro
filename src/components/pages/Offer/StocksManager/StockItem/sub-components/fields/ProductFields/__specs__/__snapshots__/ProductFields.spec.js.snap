// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Offer | StocksManager | StockItem | sub-components | fields | ProductFields should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductFields
    assignModalConfig={[MockFunction]}
    beginningDatetime="2019-01-01"
    closeInfo={[MockFunction]}
    hasIban={false}
    isEvent={false}
    readOnly={false}
    showInfo={[MockFunction]}
    stock={
      Object {
        "available": 2,
        "remainingQuantity": 1,
      }
    }
    timezone="UTC"
    venue={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <td
          title="Gratuit si vide"
        >
          <HiddenField
            name="offerId"
            type="hidden"
            validator={[Function]}
          />
          <HiddenField
            name="venueId"
            type="hidden"
            validator={[Function]}
          />
          <NumberField
            format={[Function]}
            name="price"
            onBlur={[Function]}
            placeholder="Gratuit"
            readOnly={false}
            title="Prix"
          />
        </td>,
        <td
          className="tooltiped"
          title="Laissez vide si pas de limite"
        >
          <DateField
            autoComplete={false}
            className=""
            clearable={false}
            dateFormat="DD/MM/YYYY"
            datePickerClassName="date"
            disabled={false}
            id={null}
            label=""
            locale="fr"
            name="bookingLimitDatetime"
            placeholder="Laissez vide si pas de limite"
            readOnly={false}
            renderValue={[Function]}
            required={false}
            timezone="UTC"
            validate={null}
          />
        </td>,
        <td
          className="tooltiped"
        >
          <NumberField
            format={[Function]}
            name="available"
            placeholder="Illimité"
            readOnly={false}
            renderValue={[Function]}
            title="Stock[ou] Place[s] affecté[es]"
          />
        </td>,
        <td
          className="is-small remaining-stock"
          id="remaining-stock"
        >
          1
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HiddenField
              name="offerId"
              type="hidden"
              validator={[Function]}
            />,
            <HiddenField
              name="venueId"
              type="hidden"
              validator={[Function]}
            />,
            <NumberField
              format={[Function]}
              name="price"
              onBlur={[Function]}
              placeholder="Gratuit"
              readOnly={false}
              title="Prix"
            />,
          ],
          "title": "Gratuit si vide",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "offerId",
              "type": "hidden",
              "validator": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "venueId",
              "type": "hidden",
              "validator": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "format": [Function],
              "name": "price",
              "onBlur": [Function],
              "placeholder": "Gratuit",
              "readOnly": false,
              "title": "Prix",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <DateField
            autoComplete={false}
            className=""
            clearable={false}
            dateFormat="DD/MM/YYYY"
            datePickerClassName="date"
            disabled={false}
            id={null}
            label=""
            locale="fr"
            name="bookingLimitDatetime"
            placeholder="Laissez vide si pas de limite"
            readOnly={false}
            renderValue={[Function]}
            required={false}
            timezone="UTC"
            validate={null}
          />,
          "className": "tooltiped",
          "title": "Laissez vide si pas de limite",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoComplete": false,
            "className": "",
            "clearable": false,
            "dateFormat": "DD/MM/YYYY",
            "datePickerClassName": "date",
            "disabled": false,
            "id": null,
            "label": "",
            "locale": "fr",
            "maxDate": undefined,
            "name": "bookingLimitDatetime",
            "placeholder": "Laissez vide si pas de limite",
            "readOnly": false,
            "renderValue": [Function],
            "required": false,
            "timezone": "UTC",
            "validate": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NumberField
            format={[Function]}
            name="available"
            placeholder="Illimité"
            readOnly={false}
            renderValue={[Function]}
            title="Stock[ou] Place[s] affecté[es]"
          />,
          "className": "tooltiped",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "format": [Function],
            "name": "available",
            "placeholder": "Illimité",
            "readOnly": false,
            "renderValue": [Function],
            "title": "Stock[ou] Place[s] affecté[es]",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1,
          "className": "is-small remaining-stock",
          "id": "remaining-stock",
        },
        "ref": null,
        "rendered": 1,
        "type": "td",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <td
            title="Gratuit si vide"
          >
            <HiddenField
              name="offerId"
              type="hidden"
              validator={[Function]}
            />
            <HiddenField
              name="venueId"
              type="hidden"
              validator={[Function]}
            />
            <NumberField
              format={[Function]}
              name="price"
              onBlur={[Function]}
              placeholder="Gratuit"
              readOnly={false}
              title="Prix"
            />
          </td>,
          <td
            className="tooltiped"
            title="Laissez vide si pas de limite"
          >
            <DateField
              autoComplete={false}
              className=""
              clearable={false}
              dateFormat="DD/MM/YYYY"
              datePickerClassName="date"
              disabled={false}
              id={null}
              label=""
              locale="fr"
              name="bookingLimitDatetime"
              placeholder="Laissez vide si pas de limite"
              readOnly={false}
              renderValue={[Function]}
              required={false}
              timezone="UTC"
              validate={null}
            />
          </td>,
          <td
            className="tooltiped"
          >
            <NumberField
              format={[Function]}
              name="available"
              placeholder="Illimité"
              readOnly={false}
              renderValue={[Function]}
              title="Stock[ou] Place[s] affecté[es]"
            />
          </td>,
          <td
            className="is-small remaining-stock"
            id="remaining-stock"
          >
            1
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <HiddenField
                name="offerId"
                type="hidden"
                validator={[Function]}
              />,
              <HiddenField
                name="venueId"
                type="hidden"
                validator={[Function]}
              />,
              <NumberField
                format={[Function]}
                name="price"
                onBlur={[Function]}
                placeholder="Gratuit"
                readOnly={false}
                title="Prix"
              />,
            ],
            "title": "Gratuit si vide",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "offerId",
                "type": "hidden",
                "validator": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "venueId",
                "type": "hidden",
                "validator": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "format": [Function],
                "name": "price",
                "onBlur": [Function],
                "placeholder": "Gratuit",
                "readOnly": false,
                "title": "Prix",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <DateField
              autoComplete={false}
              className=""
              clearable={false}
              dateFormat="DD/MM/YYYY"
              datePickerClassName="date"
              disabled={false}
              id={null}
              label=""
              locale="fr"
              name="bookingLimitDatetime"
              placeholder="Laissez vide si pas de limite"
              readOnly={false}
              renderValue={[Function]}
              required={false}
              timezone="UTC"
              validate={null}
            />,
            "className": "tooltiped",
            "title": "Laissez vide si pas de limite",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoComplete": false,
              "className": "",
              "clearable": false,
              "dateFormat": "DD/MM/YYYY",
              "datePickerClassName": "date",
              "disabled": false,
              "id": null,
              "label": "",
              "locale": "fr",
              "maxDate": undefined,
              "name": "bookingLimitDatetime",
              "placeholder": "Laissez vide si pas de limite",
              "readOnly": false,
              "renderValue": [Function],
              "required": false,
              "timezone": "UTC",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NumberField
              format={[Function]}
              name="available"
              placeholder="Illimité"
              readOnly={false}
              renderValue={[Function]}
              title="Stock[ou] Place[s] affecté[es]"
            />,
            "className": "tooltiped",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "format": [Function],
              "name": "available",
              "placeholder": "Illimité",
              "readOnly": false,
              "renderValue": [Function],
              "title": "Stock[ou] Place[s] affecté[es]",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "is-small remaining-stock",
            "id": "remaining-stock",
          },
          "ref": null,
          "rendered": 1,
          "type": "td",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
